src/modules/fancyzones/lib/MonitorWorkAreaHandler.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+interface IZoneWindow;
+
+namespace std
+{
+    template<>
+    struct hash<GUID>
+    {
+        size_t operator()(const GUID& Value) const
+        {
+            RPC_STATUS status = RPC_S_OK;
+            return ::UuidHash(&const_cast<GUID&>(Value), &status);
+        }
+    };
+}
+
+class MonitorWorkAreaHandler
+{
+public:
+    /**
+     * Get work area based on virtual desktop id and monitor handle.
+     *
+     * @param[in]  desktopId Virtual desktop identifier.
+     * @param[in]  monitor   Monitor handle.
+     *
+     * @returns    Object representing single work area, interface to all actions available on work area
+     *             (e.g. moving windows through zone layout specified for that work area).
+     */
+    IZoneWindow* GetWorkArea(const GUID& desktopId, HMONITOR monitor);
+
+    /**
+     * Get work area on which specified window is located.
+     *
+     * @param[in]  window Window handle.
+     *
+     * @returns    Object representing single work area, interface to all actions available on work area
+     *             (e.g. moving windows through zone layout specified for that work area).
+     */
+    IZoneWindow* GetWorkArea(HWND window);
+
+    /**
+     * Get map of all work areas on single virtual desktop. Key in the map is monitor handle, while value
+     * represents single work area.
+     *
+     * @param[in]  desktopId Virtual desktop identifier.
+     *
+     * @returns    Map containing pairs of monitor and work area for that monitor (within same virtual desktop).
+     */
+    std::unordered_map<HMONITOR, IZoneWindow*> GetWorkAreasByDesktopId(const GUID& desktopId);
+
+    /**
+     * @returns    All registered work areas.
+     */
+    std::vector<IZoneWindow*> GetAllWorkAreas();
+
+    /**
+     * Register new work area.
+     *
+     * @param[in]  desktopId Virtual desktop identifier.
+     * @param[in]  monitor   Monitor handle.
+     * @param[in]  workAra   Object representing single work area.
+     */
+    void AddWorkArea(const GUID& desktopId, HMONITOR monitor, winrt::com_ptr<IZoneWindow>& workArea);
+
+    /**
+     * Check if work area is already registered.
+     *
+     * @param[in]  desktopId Virtual desktop identifier.
+     * @param[in]  monitor   Monitor handle.
+     *
+     * @returns    Boolean indicating whether work area defined by virtual desktop id and monitor is already registered.
```

Comment:
Misleading description, by reading this I would assume that this returns true if `(desktopId, monitor)` already exists. Same could be said about "Check if work area is already registered" above.

Reply:
Since I've already described previously what is uniquely defining single work area, I guess it would be sufficient to say "if wok area is already registered" without any additional comment.